src/Symfony/Bundle/FrameworkBundle/Tests/Functional/ConfigDebugCommandTest.php

```diff
@@ -23,114 +23,154 @@
 
         $this->assertSame(0, $ret, 'Returns 0 in case of success');
         $this->assertStringContainsString('enabled: true', $tester->getDisplay());
     }
 
-    public function testDumpUndefinedBundleOption()
+    /**
+     * @dataProvider provideDebugValues
+     */
+    public function testDumpUndefinedBundleOption(bool $debug)
     {
-        $tester = $this->createCommandTester();
+        $tester = $this->createCommandTester($debug);
         $tester->execute(['name' => 'TestBundle', 'path' => 'foo']);
 
         $this->assertStringContainsString('Unable to find configuration for "test.foo"', $tester->getDisplay());
     }
 
-    public function testDumpWithPrefixedEnv()
+    /**
+     * @dataProvider provideDebugValues
+     */
+    public function testDumpWithPrefixedEnv(bool $debug)
     {
-        $tester = $this->createCommandTester();
+        $tester = $this->createCommandTester($debug);
         $tester->execute(['name' => 'FrameworkBundle']);
 
         $this->assertStringContainsString("cookie_httponly: '%env(bool:COOKIE_HTTPONLY)%'", $tester->getDisplay());
     }
 
-    public function testDumpFallsBackToDefaultConfigAndResolvesParameterValue()
+    /**
+     * @dataProvider provideDebugValues
+     */
+    public function testDumpFallsBackToDefaultConfigAndResolvesParameterValue(bool $debug)
     {
-        $tester = $this->createCommandTester();
+        $tester = $this->createCommandTester($debug);
         $ret = $tester->execute(['name' => 'DefaultConfigTestBundle']);
 
         $this->assertSame(0, $ret, 'Returns 0 in case of success');
         $this->assertStringContainsString('foo: bar', $tester->getDisplay());
     }
 
-    public function testDumpFallsBackToDefaultConfigAndResolvesEnvPlaceholder()
+    /**
+     * @dataProvider provideDebugValues
+     */
+    public function testDumpFallsBackToDefaultConfigAndResolvesEnvPlaceholder(bool $debug)
     {
-        $tester = $this->createCommandTester();
+        $tester = $this->createCommandTester($debug);
         $ret = $tester->execute(['name' => 'DefaultConfigTestBundle']);
 
         $this->assertSame(0, $ret, 'Returns 0 in case of success');
         $this->assertStringContainsString("baz: '%env(BAZ)%'", $tester->getDisplay());
     }
 
-    public function testDumpThrowsExceptionWhenDefaultConfigFallbackIsImpossible()
+    /**
+     * @dataProvider provideDebugValues
+     */
+    public function testDumpThrowsExceptionWhenDefaultConfigFallbackIsImpossible(bool $debug)
     {
         $this->expectException(\LogicException::class);
         $this->expectExceptionMessage('The extension with alias "extension_without_config_test" does not have configuration.');
 
-        $tester = $this->createCommandTester();
+        $tester = $this->createCommandTester($debug);
         $tester->execute(['name' => 'ExtensionWithoutConfigTestBundle']);
     }
 
+    public static function provideDebugValues(): iterable
+    {
+        yield [true];
+        yield [false];
```

Comment:
let's replace this provide with `@testWith` annotations

Reply:
Done https://github.com/symfony/symfony/compare/73871a75d718a38ee9204ed4918579ea14162b3b..2824fc5d69e874a625e91bad56cbbf429469dc89